src/main/java/org/broadinstitute/hellbender/engine/ReadShard.java

```diff
@@ -0,0 +1,182 @@
+     */
+    public SimpleInterval getInterval() {
+        return interval;
+    }
+
+    /**
+     * @return the interval this shard spans, potentially with additional padding on each side
+     */
+    public SimpleInterval getPaddedInterval() {
+        return paddedInterval;
+    }
+
+    /**
+     * @return an iterator over reads in this shard, as filtered using the configured read filter;
+     *         reads are lazily loaded rather than pre-loaded
+     */
+    @Override
+    public Iterator<GATKRead> iterator() {
+        final Iterator<GATKRead> readsIterator = readsSource.query(paddedInterval);
+
+        return readFilter != null ? new ReadFilteringIterator(readsIterator, readFilter) : readsIterator;
+    }
+
+    /**
+     * @return a List containing all reads in this shard, pre-loaded, and filtered using the configured read filter
+     *
+     * note: call {@link #iterator} instead to avoid pre-loading all reads at once
+     */
+    public List<GATKRead> loadAllReads() {
+        return StreamSupport.stream(Spliterators.spliteratorUnknownSize(iterator(), Spliterator.ORDERED), false).collect(Collectors.toList());
+    }
+
+    /**
+     * @param loc Locatable to test
+     * @return true if loc is contained within this shard's interval, otherwise false
+     */
+    public boolean isContainedWithinShard( final Locatable loc ) {
+        return interval.contains(loc);
+    }
+
+    /**
+     * @param loc Locatable to test
+     * @return true if loc starts within this shard's interval, otherwise false
+     */
+    public boolean startsWithinShard( final Locatable loc ) {
+        return interval.contains(new SimpleInterval(loc.getContig(), loc.getStart(), loc.getStart()));
+    }
+
+    /**
+     * @return number of bases of padding to the left of our interval
+     */
+    public int numLeftPaddingBases() {
+        return interval.getStart() - paddedInterval.getStart();
+    }
+
+    /**
+     * @return number of bases of padding to the right of our interval
+     */
+    public int numRightPaddingBases() {
+        return paddedInterval.getEnd() - interval.getEnd();
+    }
+
+    /**
+     * Shard an interval into ReadShards using the shardSize and shardPadding arguments
+     *
+     * @param interval interval to shard; must be on the contig according to the provided dictionary
+     * @param shardSize desired shard size; intervals larger than this will be divided into shards of up to this size
+     * @param shardPadding desired shard padding; each shard's interval will be padded on both sides by this number of bases
+     * @param readsSource data source for reads
+     * @param dictionary sequence dictionary for reads
+     * @return List of {@link ReadShard} objects, sharded and padded as necessary
+     */
+    public static List<ReadShard> shardInterval( final SimpleInterval interval, final int shardSize, final int shardPadding, final ReadsDataSource readsSource, final SAMSequenceDictionary dictionary ) {
```

Comment:
@droazen, could you add another parameter called `shardStep`? It will be useful for #1708, and you can implement another static method that use `shardSize` as `shardSize` for the same behaviour...

Only needs a chek that `shardStep` is not higher than `shardSize` and change line 177 by `start += shardStep`.